src/routes/(console)/(migration-wizard)/wizard.svelte

```diff
@@ -1,52 +1,409 @@
+    import { Dependencies } from '$lib/constants';
 
     const onExit = () => {
         formData.reset();
         requestedMigration.set(null);
     };
 
+    const organizations = page.data.organizations
+        .teams as Models.TeamList<Models.Preferences>['teams'];
+
+    let showExitModal = false;
+    let showResources = false;
+    let loadingProjects = false;
+    let creatingProject = false;
+    let errorInResources = false;
+    let migrationStarted = false;
+    let projectSdkInstance: ReturnType<typeof sdk.forProject> | null = null;
+
+    let projects = [] as Models.ProjectList['projects'];
+    let selectedOrg = organizations.length ? organizations[0].$id : null;
+
+    let newProjName = '';
+    let projectType: 'existing' | 'new' = 'existing';
+
+    async function getProjects(orgId: string | null) {
+        if (!orgId) {
+            projects = [];
+        } else {
+            loadingProjects = true;
+            projects = await sdk.forConsole.projects
+                .list([Query.equal('teamId', orgId), Query.orderDesc('$createdAt')])
+                .then((res) => res.projects);
+
+            loadingProjects = false;
+            projectType = projects.length ? 'existing' : 'new';
+            if (projectType === 'existing') {
+                const first = projects[0];
+                $selectedProject = first.$id;
+                projectSdkInstance = sdk.forProject(first.region, first.region);
+            }
+        }
+    }
+
+    function getProjectName(): string {
+        return isExisting ? getSelectedProject().name : newProjName || 'Appwrite project';
+    }
+
+    function getSelectedProject() {
+        return projects.find((project) => project.$id === $selectedProject);
+    }
+
+    async function createNewProject() {
+        creatingProject = true;
+
+        try {
+            return await sdk.forConsole.projects.create(
+                ID.unique(),
+                newProjName,
+                selectedOrg,
+                $selectedRegion
+            );
+        } catch (error) {
+            addNotification({
+                type: 'error',
+                message: error.message
+            });
+
+            return null;
+        } finally {
+            creatingProject = false;
+        }
+    }
+
+    function getRegions() {
+        return $regions.regions
+            .filter((region) => region.$id !== 'default')
+            .sort((regionA, regionB) => {
+                if (regionA.disabled && !regionB.disabled) {
+                    return 1;
+                }
+                return regionA.name > regionB.name ? 1 : -1;
+            })
+            .map((region) => {
+                return {
+                    label: region.name,
+                    value: region.$id,
+                    leadingHtml: `<img src='${getFlagUrl(region.flag)}' alt='Region flag'/>`,
+                    disabled: region.disabled || !region.available
+                };
+            });
+    }
```

Comment:
Not urgent, but can we abstract this code, I've seen the same function in multiple places